ReferenceAdapter/src/main/java/com/chartboost/mediation/referenceadapter/sdk/ReferenceSettings.kt

```diff
@@ -0,0 +1,60 @@
+package com.chartboost.mediation.referenceadapter.sdk
+
+import android.content.Context
+import java.lang.ref.WeakReference
+
+class ReferenceSettings private constructor(context: Context) {
+    companion object {
+        const val REFERENCE_ADAPTER_SETTINGS = "REFERENCE_ADAPTER_SETTINGS"
+        const val REFERENCE_ADAPTER_INIT_STATUS = "REFERENCE_ADAPTER_INIT_STATUS"
+        const val REFERENCE_ADAPTER_TOKEN_FETCH_STATUS = "REFERENCE_ADAPTER_TOKEN_FETCH_STATUS"
+        const val REFERENCE_ADAPTER_AD_LOAD_STATUS = "REFERENCE_ADAPTER_AD_LOAD_STATUS"
+        const val REFERENCE_ADAPTER_AD_SHOW_STATUS = "REFERENCE_ADAPTER_AD_SHOW_STATUS"
+        const val REFERENCE_ADAPTER_AD_INVALIDATE_STATUS = "REFERENCE_ADAPTER_AD_INVALIDATE_STATUS"
+
+        @Volatile
+        private var INSTANCE: ReferenceSettings? = null
+
+        fun getInstance(context: Context): ReferenceSettings {
+            return INSTANCE ?: synchronized(this) {
+                INSTANCE ?: ReferenceSettings(context.applicationContext).also { INSTANCE = it }
+            }
+        }
+    }
+
+    private val weakContext = WeakReference(context)
```

Comment:
and you don't have to do something like this. storing app context is always okay. so just store `private val context = context.applicationContext` if you need it

Reply:
Good call. I wasn't thinking of application Context. New commit up to simplify all this.